File: src/main/java/net/atos/zac/util/UriUtil.java
/*
 * SPDX-FileCopyrightText: 2021 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.zac.util;

import static org.apache.commons.lang3.StringUtils.contains;
import static org.apache.commons.lang3.StringUtils.substringAfterLast;

import java.net.URI;
import java.util.UUID;


public final class UriUtil {

    private UriUtil() {
    }

    public static UUID uuidFromURI(final URI uri) {
        return uuidFromURI(uri.getPath());
    }

    public static UUID uuidFromURI(final String uri) {
        return UUID.fromString(extractUUID(uri));
    }

    public static boolean isEqual(final URI a, final URI b) {
        return (a != null && b != null) ?
                extractUUID(a.getPath()).equals(extractUUID(b.getPath())) :
                a == null && b == null;
    }

    private static String extractUUID(final String path) {
        return contains(path, "/") ? substringAfterLast(path, "/") : path;
    }
}

File: src/main/kotlin/net/atos/zac/util/UriUtil.kt
/*
 * SPDX-FileCopyrightText: 2021 Atos, 2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.util

import org.apache.commons.lang3.StringUtils
import java.net.URI
import java.util.UUID

fun uuidFromURI(uri: URI): UUID = uuidFromURI(uri.getPath())

fun uuidFromURI(uri: String): UUID = UUID.fromString(extractUUID(uri))

fun isEqual(a: URI?, b: URI?): Boolean =
    if (a != null && b != null) (extractUUID(a.getPath()) == extractUUID(b.getPath())) else a == null && b == null

private fun extractUUID(path: String): String =
    if (StringUtils.contains(path, "/")) StringUtils.substringAfterLast(path, "/") else path
